spec/models/product_spec.rb

```diff
@@ -115,4 +115,24 @@
       expect(level_one_extension.mirrored_extensions.for_root_product(base_b)).to eq([level_two_extension_b_mirrored])
     end
   end
+
+  describe '#recommended_for?' do
+    let!(:base_a) { FactoryGirl.create(:product) }
+    let!(:base_b) { FactoryGirl.create(:product) }
+
+    it 'recommended for base product A' do
+      extension = FactoryGirl.create(:product, :extension, base_products: [base_a], recommended: true)
+      expect(extension.recommended_for?(base_a)).to be(true)
+    end
+
+    it 'not recommended for base product A' do
+      extension = FactoryGirl.create(:product, :extension, base_products: [base_a], recommended: false)
+      expect(extension.recommended_for?(base_a)).to be(false)
+    end
+
+    it 'not recommended for unrelated product B' do
+      extension = FactoryGirl.create(:product, :extension, base_products: [base_a], recommended: true)
+      expect(extension.recommended_for?(base_b)).to be(false)
+    end
+  end
```

Comment:
To follow the http://www.betterspecs.org/ convention, I would write these tests like this.
(Take it or leave it 😄)

```ruby
describe '#recommended_for?' do
  let(:base) { create :product }
  let(:extension) { create(:product, :extension, base_products: [base], recommended: recommended) }
  
  subject { extension.recommended_for?(queried_base) }
  
  context 'when the extension is recommended for its base' do
    let(:recommended) { true }
    let(:queried_base) { base }
    
    it { is_expected.to be true }
  end

  context 'when the extension is not recommended for its base' do
    let(:recommended) { false }
    let(:queried_base) { base }
    
    it { is_expected.to be false }
  end

  context "when the queried base is not the extension's base" do
    let(:recommended) { true }
    let(:queried_base) { create(:product) }
    
    it { is_expected.to be false }
  end
end
```

Reply:
Yep, I definitely like it! Just commit those changes yourself, maybe?